blood/src/main/java/cz/cuni/mff/d3s/blood/dependencyMatrix/DependencyMatrixCollector.java

```diff
@@ -1,36 +1,56 @@
 package cz.cuni.mff.d3s.blood.dependencyMatrix;
 
+import cz.cuni.mff.d3s.blood.utils.CheckedConsumer;
 import cz.cuni.mff.d3s.blood.utils.Result;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
 import org.graalvm.compiler.graph.Node;
 import org.graalvm.compiler.nodes.StructuredGraph;
 
 import java.io.PrintWriter;
 import java.io.StringWriter;
 import java.lang.reflect.Field;
 import java.util.Arrays;
+import java.util.Date;
+import java.util.HashSet;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.graalvm.compiler.phases.BasePhase;
 
 public final class DependencyMatrixCollector {
-    static ConcurrentHashMap<Class, ConcurrentHashMap<Class, DependencyValue>> dependencyTable = new ConcurrentHashMap<>();
+
+    private static final Predicate<Object> NON_NULL = Predicate.isEqual(null).negate();
```

Comment:
This is weird. Because it's used only once, I woudn't expose it as a constant. My argument would be, that code is already constant and there is no need in doing it through a variable.

Reply:
I originally used it multiple times. I was missing something like this from the standard library. I will (re)move it.